### PR TITLE
[master] support public readonly properties, fixes #27

### LINKED ISSUE #27: PHP 8.1: Unable to load classes with public readonly properties
When loading a class with readonly properties, an error is thrown: `PHP Fatal error:  Uncaught Error: Cannot initialize readonly property ...`

This happens, because readonly properties 

### TestCase:
```php
<?php

require 'vendor/autoload.php';

use Brick\VarExporter\VarExporter;

class Test
{
    public function __construct(
        public readonly string $foo
    ) {
    }
}

$object = new Test('bar');
$string = VarExporter::export($object, VarExporter::ADD_RETURN);
echo $string;

$newObject = eval($string);
echo "yay!";
```
#### Output
``` 
return (static function() {
    $class = new \ReflectionClass(\Test::class);
    $object = $class->newInstanceWithoutConstructor();

    $object->foo = 'bar';

    return $object;
})();

Fatal error: Uncaught Error: Cannot initialize readonly property Test::$foo from global scope in  test.php(19) : eval()'d code on line 5

Error: Cannot initialize readonly property Test::$foo from global scope in test.php(19) : eval()'d code on line 5
```
#### Expected Output
```
return (static function() {
    $class = new \ReflectionClass(\Test::class);
    $object = $class->newInstanceWithoutConstructor();
    (function() {
        $this->foo = 'bar';
    })->bindTo($object, \Test::class)();

    return $object;
})();yay!
```
### Possible solution
instead of just setting the property on the object via `$object->foo = 'bar'` it should be set via `bindTo`:
```php
(function() {
        $this->foo = 'bar';
    })->bindTo($object, \Test::class)();
```
one could use `$property-> isReadOnly` to determine if this is necessary.

### PR DESCRIPTION
**Pull request by AnnaDamm**
See #27

### CODE REVIEW

**Review by func0der — APPROVED** [review]

**Review by pascalheidmann — APPROVED** [review]

**Comment by AnnaDamm** [comment]
@BenMorel Could you have a look at this PR, please? :-)

File: tests/Classes/PublicReadonlyPropertiesWithConstructor.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brick\VarExporter\Tests\Classes;
+
+class PublicReadonlyPropertiesWithConstructor
```
Comment (BenMorel) [inline tests/Classes/PublicReadonlyPropertiesWithConstructor.php:7]: Shouldn't this be:

```suggestion
class ReadonlyPropertiesWithConstructor
```

As you have private readonly as well?

**Review by BenMorel — CHANGES_REQUESTED** [review]
Thank you for this PR! 🫶

**Comment by BenMorel** [comment]
@AnnaDamm Do you have time to apply suggestions? Do you want me to take over?

File: src/Internal/ObjectExporter/AnyObjectExporter.php [hunk 2]
```diff
@@ -44,9 +44,9 @@ public function export(object $object, \ReflectionObject $reflectionObject, arra
         $returnNewObject = ($reflectionObject->getConstructor() === null);
 
         while ($current) {
-            $publicProperties = [];
+            $publicOrReadonlyProperties = [];
```
[the remark this answers is not in the document]
Comment (AnnaDamm) [reply src/Internal/ObjectExporter/AnyObjectExporter.php:47]: It is possible to have private readonly properties (check the test case for that), so they don't have to be public. That is why I chose "or"

[the remark this answers is not in the document]
Comment (AnnaDamm) [reply src/Internal/ObjectExporter/AnyObjectExporter.php:47]: Yes you are right

**Comment by AnnaDamm** [comment]
Hey @BenMorel! Sorry, I have been on vacation over the past 2 weeks and will be able to change the PR next week

**Comment by BenMorel** [comment]
Thank you, @AnnaDamm !

**Comment by BenMorel** [comment]
Released as [0.3.8](https://github.com/brick/varexporter/releases/tag/0.3.8).

**Pull request merged**